ollama/_types.py

```diff
@@ -23,7 +23,30 @@ def __setitem__(self, key: str, value: Any) -> None:
     setattr(self, key, value)
 
   def __contains__(self, key: str) -> bool:
-    return hasattr(self, key)
+    """
+    >>> msg = Message(role='user')
+    >>> 'nonexistent' in msg
+    False
+    >>> 'role' in msg
+    True
+    >>> 'content' in msg
+    False
+    >>> msg.content = 'hello!'
+    >>> 'content' in msg
+    True
+    >>> msg = Message(role='user', content='hello!')
+    >>> 'content' in msg
+    True
+    >>> 'tool_calls' in msg
+    False
+    >>> msg['tool_calls'] = []
+    >>> 'tool_calls' in msg
+    True
+    >>> msg['tool_calls'] = [Message.ToolCall(function=Message.ToolCall.Function(name='foo', arguments={}))]
+    >>> 'tool_calls' in msg
+    True
+    """
+    return key in self.model_fields_set
```

Comment:
Will this inadvertently filter out fields with a default value?

Reply:
I don't think so

> Returns the set of fields that have been explicitly set on this model instance.

Reply:
Did we want this?

```
return key in self.model_fields_set or key is not None
```

I don't think default values are in `model_fields_set`

```
@property
    def model_fields_set(self) -> set[str]:
        """Returns the set of fields that have been explicitly set on this model instance.

        Returns:
            A set of strings representing the fields that have been set,
                i.e. that were not filled from defaults.
        """
        return self.__pydantic_fields_set__
```

Reply:
<img width="326" alt="Screenshot 2024-11-23 at 6 03 36 PM" src="https://github.com/user-attachments/assets/58e92890-9947-4cad-9249-824fb4da675e">
Just checked - default values are omitted in this case which is not what we want.